### PR TITLE
Add Civigrant grant type and grant status mgd files and path to civi_…

### PR DESCRIPTION
**Pull request by Damilare1**
Overview
----------------------------------------
Add Civigrant grant_type and option values search mgd files, update paths, and add form templates for Grant Status and Grant Types to civicrm_admin_ui ang directory.

Before
----------------------------------------

Grant Status 
Grant status (old)

Grant type
Grant type (old)

After
----------------------------------------
Grant Status 
grant status (new)

Grant type
Grant type (new)

Technical Details
----------------------------------------
Created the new path in CRM/Core/xml/Menu/Admin.xml in order to prevent the actions from redirecting to the same base page.

### CODE REVIEW

**Comment by mattwire** [comment]
@colemanw I think that this should this be included in the civigrant extension instead of admin_ui?

File: CRM/Core/xml/Menu/Admin.xml [hunk 1]
```diff
@@ -580,7 +580,21 @@
...
      <path>civicrm/admin/options/grant_type</path>
      <title>Grant Types</title>
-     <desc>List of types which can be assigned to Grants. (Enable CiviGrant from Administer > Systme Settings > Enable Components if you want to track grants.)</desc>
+     <desc>List of types which can be assigned to Grants. (Enable CiviGrant from Administer > System Settings > Enable Components if you want to track grants.)</desc>
+     <page_callback>CRM_Admin_Page_Options</page_callback>
+     <adminGroup>Option Lists</adminGroup>
+     <weight>385</weight>
+  </item>
+    <item>
+     <path>civicrm/admin/options/grant_status/edit</path>
```
Comment (seamuslee001) [inline CRM/Core/xml/Menu/Admin.xml:589]: Should this not civi in the civigrant extension?

File: ext/civicrm_admin_ui/ang/afsearchGrantStatus.aff.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<div class="af-container">
```
Comment (seamuslee001) [inline ext/civicrm_admin_ui/ang/afsearchGrantStatus.aff.html:1]: Likewise shouldn't this move into the civigrant extension or is this extension going to cover grant stuff as well @colemanw ?

**Comment by colemanw** [comment]
I agree with the reviewers that all Grant-related code, including these new search displays and afforms, plus the existing menu xml, ought to be moved to the CiviGrant extension.

File: CRM/Core/xml/Menu/Admin.xml (hunk 1 above)
Comment (Damilare1) [reply CRM/Core/xml/Menu/Admin.xml]: thanks, this has been moved to the extension now.

File: ext/civicrm_admin_ui/ang/afsearchGrantStatus.aff.html (hunk 2 above)
Comment (Damilare1) [reply ext/civicrm_admin_ui/ang/afsearchGrantStatus.aff.html]: this too has been moved to the extension, thanks for the review.

**Comment by Damilare1** [comment]
Thanks @mattwire @colemanw and @seamuslee001. the suggested changes have been added.

File: ext/civigrant/ang/afsearchGrantStatus.aff.html [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<div class="af-container">
+  <div class="af-markup">
+    <div class="help">
+      <p>
+        {{:: ts('The existing option choices for Grant status group are listed
+        below. You can add, edit or delete them from this screen.') }}
```
Comment (colemanw) [inline ext/civigrant/ang/afsearchGrantStatus.aff.html:6]: Please make this a single line.

File: ext/civigrant/xml/Menu/Grant.xml [hunk 4]
```diff
@@ -35,4 +35,18 @@
...
+     <adminGroup>Option Lists</adminGroup>
+     <weight>385</weight>
+  </item>
+  <item>
+     <path>civicrm/admin/options/grant_type/edit</path>
+     <title>Edit Grant Status</title>
+     <page_callback>CRM_Admin_Page_Options</page_callback>
+     <adminGroup>Option Lists</adminGroup>
+     <weight>385</weight>
+  </item>
```
Comment (colemanw) [inline ext/civigrant/xml/Menu/Grant.xml:51]: These additions look good, but they also need to be removed from `Admin.xml`.

File: CRM/Admin/Page/Options.php [hunk 5]
```diff
@@ -152,6 +152,12 @@ public function getBAOName() {
   public function &links() {
     if (!self::$_links) {
       self::$_links = [
+        CRM_Core_Action::VIEW => [
+          'name' => ts('View'),
+          'url' => 'civicrm/admin/options/' . self::$_gName,
+          'qs' => 'action=view&id=%%id%%&reset=1',
+          'title' => ts('View Relationship Type'),
+        ],
```
Comment (colemanw) [inline CRM/Admin/Page/Options.php:160]: I think this change doesn't belong in this PR.

File: CRM/Core/xml/Menu/Admin.xml [hunk 6]
```diff
@@ -580,7 +580,7 @@
   <item>
      <path>civicrm/admin/options/grant_type</path>
      <title>Grant Types</title>
-     <desc>List of types which can be assigned to Grants. (Enable CiviGrant from Administer > Systme Settings > Enable Components if you want to track grants.)</desc>
+     <desc>List of types which can be assigned to Grants. (Enable CiviGrant from Administer > System Settings > Enable Components if you want to track grants.)</desc>
```
Comment (colemanw) [inline CRM/Core/xml/Menu/Admin.xml:583]: This change has already been merged to core in your other PR, so if you rebase this one it will disappear:
`git pull --rebase origin master`

File: ext/civigrant/xml/Menu/Grant.xml (hunk 4 above)
Comment (Damilare1) [reply ext/civigrant/xml/Menu/Grant.xml:51]: Thanks @colemanw, I think I removed this paths from Admin.xml in the RelationshipType PR before it was merged.

Comment (colemanw) [reply ext/civigrant/xml/Menu/Grant.xml:51]: No it's still there in master:
https://github.com/civicrm/civicrm-core/blob/f0f09be1f7e59f4ad44a44fee5a67276fafb7678/CRM/Core/xml/Menu/Admin.xml#L590-L597

Comment (Damilare1) [reply ext/civigrant/xml/Menu/Grant.xml:59]: ohh this wasn't added by the RelationshipType PR. I just took removed it in the latest commit.

**Comment by colemanw** [comment]
@Damilare1 thanks for your work on this PR. I'm in the process of testing it and also thinking about how it fits into the larger picture of **all** our OptionGroup admin screens. I'm going leave this open as a draft while I ponder all of that..

**Comment by colemanw** [comment]
This is going to be rolled into https://github.com/civicrm/civicrm-core/pull/25387
